lib/ModuleFilenameHelpers.js

```diff
@@ -141,7 +141,7 @@ ModuleFilenameHelpers.createFilename = (
 		absoluteResourcePath = () => module.identifier().split("!").pop();
 		hash = getHash(identifier, hashFunction);
 	}
-	const resource = memoize(() => shortIdentifier().split("!").pop());
+	const resource = memoize(() => shortIdentifier().split("|").pop());
```

Comment:
I guess getting resource from identifier isn't the best idea at all.

Instead you could get `requestShortener.shorten(module.nameForCondition())` when `module` is not a string.

For the `typeof module === "string"` case we can just assume that `split("!")` works fine

Reply:
Added the fix for `absoluteResourcePath` too, which is the one I really need :).